certstest/searcher.go

```diff
@@ -59,14 +54,6 @@ func (s *Searcher) SearchAppOperator(clusterID string) (certs.AppOperator, error
 	return s.appOperator, nil
 }
 
-func (s *Searcher) SearchCluster(clusterID string) (certs.Cluster, error) {
-	if s.clusterError != nil {
-		return certs.Cluster{}, s.clusterError
-	}
-
-	return s.cluster, nil
-}
```

Comment:
I was thinking how to make the primitives composable. There wasn't a really cool way to do it. So what I thought we should do now is to compose separately within the infrastructure providers aka `aws-operator`. That gives us the flexibility to decide which Etcd certs to fetch and dispatch per Tenant Cluster. That implies that I would like to drop the notion of a "cluster" here, as nobody can really tell what a "cluster" then is. Further changes we want to do tomorrow should be fine because we use the most granular version and compose as we like. A method to do that already exists, which is `Searcher.SearchTLS`. Since with this the interface changes quite a lot I would like to get approval from @tuommaki and @MarcelMue as well. So that we have all provider expertise in here.

Reply:
I'm fine with doing the composing in provider operators, but would it make sense to replace `SearchCluster()` with similar helper: `func (s *Searcher) SearchTLSs(clusterID string, certs []certs.Cert) ([]certs.TLS, error)`

That's what I'd probably implement in the provider operator so one can then just define list of certs that are interesting and get them at once.